### ISSUE TITLE
Unable to auth with Notion

### ISSUE THREAD

**Issue by eob** [excalidraw/excalidraw#4233]
Hello!

I just signed up for a pro account so we can start embedding Excalidraw images in Notion. 

Here is the sequence of steps to reproduce my error:
1. Paste a private Excalidraw URL into Notion
2. Notion displays an embedded Excalidraw "Click on the Button to Log In" frame
3. I click on Google
4. I select the Google account I used with Excalidraw
5. The auth screen shows white and hangs forever
6. If I refresh the page, I see `Unable to process request due to missing initial state. This may happen if browser sessionStorage is inaccessible or accidentally cleared.`

As a result I can't embed images.

As an alternative, I tried enabling link-sharing, but then the embedded image shows up with lots of additional UI elements (displayed within notion) inviting me to sign up for Excalidraw.

**Comment by nickfreemandesign** [comment]
Yes, also experiencing this issue. Additionally, login with the manual link will route the user away from Notion, into their email, and then back into the browser. So that strategy falls short for any embedded widgets such as in Notion.

We love what you do, Excalidraw!

**Comment by dwelle** [comment]
Oops, missed to original request.

Yes, support for better sharing / embedding of scenes will come soon. We will have read-only links with the most minimal panache: no live collaboration, UI or similar. Though it will still be interactive so you can pan/zoom.

>  Additionally, login with the manual link will route the user away from Notion, into their email, and then back into the browser.

Unclear what you mean about the email step?

**Comment by nickfreemandesign** [comment]
@dwelle Heres the troublesome flow:

1. Open Notion (or wherever you are about to embed Excalidraw)
2. Create your embed by pasting an Excalidraw+ private link
3. The embed asks you to sign in.
4a. Click signin via Google, and you will get the flow that @eob describes. Google auth popup appears in your browser, and after selecting your account, the screen stays in a perpetual loading state
4b. Click signin via email, and you will receive a popup in your browser to login via email. This includes some kind of magic-link flow, but regardless the ultimate outcome is that you are signed in to your browser, and not the embed.

Hope that helps

**Comment by nickfreemandesign** [comment]
@dwelle any status here? Looking to make much more use of excalidraw+ by using it within notion

**Comment by dwelle** [comment]
Hey @nickfreemandesign — apologies for the late reply. Improved workflow will be shipping within a week!

**Comment by lawrencejones** [comment]
Just a note to say I've hit this today, by clicking "sign-in to Excalidraw" from within the Notion app and the browser hanging on the Excalidraw auth redirect once I've tried signing in.

**Comment by dwelle** [comment]
@lawrencejones note that for personal/team use cases it should work if you are all signed in already (JWT saved in browser), and it shouldn't prompt.

That said, hacky for sure. It shouldn't be long now for shipping better readonly support. (We got stuck on work elsewhere, 😥).

**Comment by paudrow** [comment]
I'm having the same problem. It works fine for me on a web browser, but when I try to use the Notion Desktop app (MacOS), I'm not able to sign in. The web browser asks me which account to use and then hangs, never getting back to my Notion Desktop app.

I'm looking forward to this being fixed.

**Comment by aantipov** [comment]
> I'm having the same problem. It works fine for me on a web browser, but when I try to use the Notion Desktop app (MacOS), I'm not able to sign in. The web browser asks me which account to use and then hangs, never getting back to my Notion Desktop app.
> 
> One possible thing to note, I'm using Brave (Chromium based) as my default browser.

Same issue. The Excalidraw embed works just fine in the browser (Chrome), but not in the MacOS Desktop App. The login issue

**Comment by tommyRmorava-hub** [comment]
Same issue here. Im able to auth via notion in browser, but when I go to app, it wants me auth again and it doesnt work. When I refresh after a while, i get this message: "Unable to process request due to missing initial state. This may happen if browser sessionStorage is inaccessible or accidentally cleared. Some specific scenarios are - 1) Using IDP-Initiated SAML SSO. 2) Using signInWithRedirect in a storage-partitioned browser environment." 

Please help!
